### PR TITLE
Fix docs init

### PR DESCRIPTION
**Pull request by saadjutt01**
## Issue

Few improvements + command alias + sub-command

## Intent

Write tests for `sasjs doc`

## Implementation

++ alias `sasjs docs`
++ subcommand `sasjs doc init`
++ updated cli message
++ tests for `sasjs doc` and `sasjs doc init`
++ to pass all tests locally, Doxygen should be installed

## Checks

- [ ] Code is formatted correctly (`npm run lint:fix`).
- [ ] Any new functionality has been unit tested.
- [ ] All unit tests are passing (`npm test`).
- [ ] All CI checks are green.
- [ ] JSDoc comments have been added or updated.
- [ ] Reviewer is assigned.

image
image

### CODE REVIEW

**Review by allanbowe — APPROVED** [review]

File: src/commands/docs/spec/generateDocs.spec.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  )
+
+  it(
+    `should generate docs for single target`,
+    async () => {
+      appName = `test-app-doc-${generateTimestamp()}`
+
+      await createTestApp(__dirname, appName)
+
+      await expect(doc(new Command(`doc -t viya`))).resolves.toEqual(0)
```
Comment (krishna-acondy) [inline src/commands/docs/spec/generateDocs.spec.ts:37]: Should we not assert on the contents of the doc folder?

File: src/main.ts [hunk 2]
```diff
@@ -46,12 +47,26 @@ export async function createFileStructure(command: Command) {
...
     })
 }
-export async function generateDocs(command: Command) {
+export async function doc(command: Command) {
+  const subCommand = command.getSubCommand()
+
+  if (subCommand === 'init') {
+    return await initDocs()
+      .then(() => {
+        displaySuccess(`Docs are initialited. You can generate docs now.`)
```
Comment (krishna-acondy) [inline src/main.ts:56]: _initialised_

Comment (allanbowe) [reply src/main.ts:56]: The doxygen configuration files have been initialised under `/sasjs/doxy/`.  You can now run `sasjs doc`.

File: src/commands/docs/spec/generateDocs.spec.ts (hunk 1 above)
Comment (saadjutt01) [reply src/commands/docs/spec/generateDocs.spec.ts:37]: Content we get:

```
saad@saad-acer:~/projects/macropeople/apps_sasjs/t1$ tree sasjsbuild/docs/
sasjsbuild/docs/
├── appinit_8sas.html
├── appinit_8sas_source.html
├── bc_s.png
├── bdwn.png
├── closed.png
├── dir_28cd64f980436b9184807de48c3cf4aa.html
├── dir_28cd64f980436b9184807de48c3cf4aa.js
├── dir_2dc29ef64dcb6013bd6e4ea806b1887e.html
├── dir_2dc29ef64dcb6013bd6e4ea806b1887e.js
├── dir_7850eb5db615769e5ae107623c01dbe7.html
├── dir_7850eb5db615769e5ae107623c01dbe7.js
├── dir_79f7e5956d06a61f44fca2d0633ffa0b.html
├── dir_79f7e5956d06a61f44fca2d0633ffa0b.js
├── dir_8093be6e57e8700097d799aa1db1478d.html
├── dir_8093be6e57e8700097d799aa1db1478d.js
├── dir_85224d19f93510b2f56c7960a02145d7.html
├── dir_85224d19f93510b2f56c7960a02145d7.js
├── dir_9d8a3c22713511026de502e4febc1f2f.html
├── dir_9d8a3c22713511026de502e4febc1f2f.js
├── dir_a1f7f05bba2c191bc662e15b7b70f292.html
├── dir_a1f7f05bba2c191bc662e15b7b70f292.js
├── dir_a86e6f7a05ed700962df54fb09824706.html
├── dir_a86e6f7a05ed700962df54fb09824706.js
├── dir_aeff632c6930a27bc2b46aa5540a6fd0.html
├── dir_aeff632c6930a27bc2b46aa5540a6fd0.js
├── dir_c5ca297606fe7c9cd0a6b53f108e6d4a.html
├── dir_c5ca297606fe7c9cd0a6b53f108e6d4a.js
├── dir_ccaa2df6696e941e426e2541f60d9407.html
├── dir_ccaa2df6696e941e426e2541f60d9407.js
├── dir_cfc38a21734c3edb6bc9dd88c6d4c406.html
├── dir_cfc38a21734c3edb6bc9dd88c6d4c406.js
├── dir_d9969a00905f54d9980d06644a33f8df.html
├── dir_d9969a00905f54d9980d06644a33f8df.js
├── dir_ef8e8a8acdb1b056d823c7b57e55db78.html
├── dir_ef8e8a8acdb1b056d823c7b57e55db78.js
├── doc.png
├── doesnothing_8sas_source.html
├── doxygen.css
├── doxygen.svg
├── dynsections.js
├── example_8sas.html
├── example_8sas_source.html
├── exampleprogram_8sas_source.html
├── favicon.ico
├── files.html
├── files.js
├── folderclosed.png
├── folderopen.png
├── getdata_8sas.html
├── getdata_8sas_source.html
├── index.html
├── jquery.js
├── macros_2examplemacro_8sas.html
├── macros_2examplemacro_8sas_source.html
├── nav_f.png
├── nav_g.png
├── nav_h.png
├── navtree.css
├── navtreedata.js
├── navtreeindex0.js
├── navtree.js
├── new_stylesheet.css
├── open.png
├── resize.js
├── sasjs-logo.png
├── search
│   ├── all_0.html
│   ├── all_0.js
│   ├── all_1.html
│   ├── all_1.js
│   ├── all_2.html
│   ├── all_2.js
│   ├── all_3.html
│   ├── all_3.js
│   ├── all_4.html
│   ├── all_4.js
│   ├── close.svg
│   ├── files_0.html
│   ├── files_0.js
│   ├── files_1.html
│   ├── files_1.js
│   ├── files_2.html
│   ├── files_2.js
│   ├── files_3.html
│   ├── files_3.js
│   ├── files_4.html
│   ├── files_4.js
│   ├── mag_sel.svg
│   ├── nomatches.html
│   ├── search.css
│   ├── searchdata.js
│   ├── search.js
│   ├── search_l.png
│   ├── search_m.png
│   └── search_r.png
├── services_2admin_2dostuff_8sas.html
├── services_2admin_2dostuff_8sas_source.html
├── splitbar.png
├── sync_off.png
├── sync_on.png
├── tab_a.png
├── tab_b.png
├── tab_h.png
├── tabs.css
├── tab_s.png
├── targets_2sas9_2macros_2examplemacro_8sas_source.html
├── targets_2sas9_2services_2admin_2dostuff_8sas_source.html
├── targets_2viya_2macros_2examplemacro_8sas_source.html
├── targets_2viya_2services_2admin_2dostuff_8sas_source.html
├── yetanothermacro_8sas.html
└── yetanothermacro_8sas_source.html

1 directory, 110 files

```

Comment (saadjutt01) [reply src/commands/docs/spec/generateDocs.spec.ts:37]: @allanbowe , any suggestion to which files needs to be check as valid docs?

Comment (allanbowe) [reply src/commands/docs/spec/generateDocs.spec.ts:37]: the index.html file should always exist

Comment (allanbowe) [reply src/commands/docs/spec/generateDocs.spec.ts:37]: we could also test for `appinit_8sas.html` to confirm that the service was actually documented

**Review by allanbowe — DISMISSED** [review]

File: src/commands/docs/spec/generateDocs.spec.ts (hunk 1 above)
Comment (saadjutt01) [reply src/commands/docs/spec/generateDocs.spec.ts:49]: added in `verifydocs` function

**Review by allanbowe — APPROVED** [review]

**Pull request merged**